KubeArmor/BPF/enforcer_path.bpf.c

```diff
@@ -0,0 +1,95 @@
+// +build ignore
+
+#include "shared.h"
+
+SEC("lsm/path_mknod")
+int BPF_PROG(enforce_mknod, struct path *dir, struct dentry *dentry) {
+  struct path f_path;
+  f_path.dentry = dentry;
+  f_path.mnt = BPF_CORE_READ(dir, mnt);
+  return match_and_enforce_path_hooks(&f_path, dpath);
+}
```

Comment:
You can have a macro that could expand to all below APIs ...

```suggestion
#define PATH_SEC_CALL(NAME) \
SEC("lsm/path_##NAME")\
int BPF_PROG(enforce_##NAME, struct path *dir, struct dentry *dentry) {
  struct path f_path;
  f_path.dentry = dentry;
  f_path.mnt = BPF_CORE_READ(dir, mnt);
  return match_and_enforce_path_hooks(&f_path, dpath);
}

PATH_SEC_CALL(mknod)
PATH_SEC_CALL(rmdir)
PATH_SEC_CALL(unlink)
:: 
```